torchvision/models/detection/_utils.py

```diff
@@ -270,6 +270,43 @@ def encode_single(self, reference_boxes: Tensor, proposals: Tensor) -> Tensor:
 
         return targets
 
+    def encode_all(self, reference_boxes: List[Tensor], proposals: List[Tensor]) -> Tensor:
+        """
+        vectorized version of `encode_single`
+        Args:
+            reference_boxes (Tensor): reference boxes
+            proposals (Tensor): boxes to be encoded
+
+        Returns:
+            Tensor: the encoded relative box offsets that can be used to
+            decode the boxes.
+
+        """
+
+        reference_boxes = torch.stack(reference_boxes)
```

Comment:
Yes, I understand your point. But that might introduce the Synchronization issue between CPU and GPU as noted in #5322 and #5691.

Reply:
@cpuhrsch Could you advise? 

Will code like `x = x if torch.is_tensor(x) else torch.stack(x)` cause extra synchronization?

In the PR above it was checking onto result of `any()` function causing GPU->CPU sync. In this case with torch.is_tensor(x) it should not be a problem IMO as dispatch is done anyway. Not sure how the code as above gets traced/scripted/fx'd.

Reply:
@vadimkantorov - I don't think expect it to, but you can use https://pytorch.org/docs/master/generated/torch.cuda.set_sync_debug_mode.html#torch.cuda.set_sync_debug_mode to find out

Reply:
> One way to avoid this would be to just accept tensor and the torch.stack(reference_boxes) and torch.stack(proposals) would be done by the caller.

IMO this would be great and lead to a cleaner API!

Reply:
Agree 💯